app_dart/test/request_handlers/check_for_waiting_pull_requests_test.dart

```diff
@@ -0,0 +1,402 @@
+      config.waitingForTreeToGoGreenLabelNameValue =
+          'waiting for tree to go green';
+      config.githubGraphQLClient = githubGraphQLClient;
+
+      handler = CheckForWaitingPullRequests(
+        config,
+        auth,
+        loggingProvider: () => log,
+      );
+    });
+
+    void _verifyQueries() {
+      githubGraphQLClient.verifyQueries(
+        <QueryOptions>[
+          QueryOptions(
+            document: labeledPullRequestsWithReviewsQuery,
+            fetchPolicy: FetchPolicy.noCache,
+            variables: <String, dynamic>{
+              'sOwner': 'flutter',
+              'sName': 'flutter',
+              'sLabelName': config.waitingForTreeToGoGreenLabelNameValue,
+            },
+          ),
+          QueryOptions(
+            document: labeledPullRequestsWithReviewsQuery,
+            fetchPolicy: FetchPolicy.noCache,
+            variables: <String, dynamic>{
+              'sOwner': 'flutter',
+              'sName': 'engine',
+              'sLabelName': config.waitingForTreeToGoGreenLabelNameValue,
+            },
+          ),
+        ],
+      );
+    }
+
+    test('Merges first PR in list, all successful', () async {
+      flutterRepoPRs.add(PullRequestHelper());
+      flutterRepoPRs.add(PullRequestHelper()); // will be ignored.
+      engineRepoPRs.add(PullRequestHelper());
+
+      await tester.get(handler);
+
+      _verifyQueries();
+
+      githubGraphQLClient.verifyMutations(
+        <MutationOptions>[
+          MutationOptions(
+            document: mergePullRequestMutation,
+            variables: <String, dynamic>{
+              'id': flutterRepoPRs.first.id,
+              'oid': 'deadbeef',
+            },
+          ),
+          MutationOptions(
+            document: mergePullRequestMutation,
+            variables: <String, dynamic>{
+              'id': engineRepoPRs.first.id,
+              'oid': 'deadbeef',
+            },
+          ),
+        ],
+      );
+    });
+
+    test('Ignores PRs that are too new', () async {
+      flutterRepoPRs.add(PullRequestHelper(dateTime: DateTime.now()));
+      flutterRepoPRs.add(PullRequestHelper()); // will be ignored.
+      engineRepoPRs.add(PullRequestHelper());
+
+      await tester.get(handler);
+
+      _verifyQueries();
+
+      githubGraphQLClient.verifyMutations(
+        <MutationOptions>[
+          MutationOptions(
+            document: mergePullRequestMutation,
+            variables: <String, dynamic>{
+              'id': flutterRepoPRs.last.id,
+              'oid': 'deadbeef',
+            },
+          ),
+          MutationOptions(
+            document: mergePullRequestMutation,
+            variables: <String, dynamic>{
+              'id': engineRepoPRs.first.id,
+              'oid': 'deadbeef',
+            },
+          ),
+        ],
+      );
+    });
+
+    test('Remove labels', () async {
+      final PullRequestHelper prOneBadReview = PullRequestHelper(
+        hasApprovedReview: false,
```

Comment:
Is there any test that covers how we parsed the reviews and set the `hasApprovedReview`, `hasChangeRequestReview`, ...,  flags?

Reply:
The `Remove Labels` test is doing that - it's basically exercising a whole bunch of scenarios we care about in those branches.

Reply:
Oh, now I see that `PullRequestHelper` is a test class that generates the corresponding string using those flags.